### PR TITLE
[SYCL] Fix post commit failure on Windows.

### PR DESCRIPTION
**Pull request by srividya-sundaram**
Fixes post-commit failure reported in https://github.com/intel/llvm/pull/7483

### CODE REVIEW

**Review by smanna12 — DISMISSED** [review]

File: clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp
Comment (premanandrao) [inline clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:13]:
```suggestion
      // CHECK: call {{.*}}printf(ptr noundef nonnull dereferenceable(1) @{{.*}}, i32 noundef 24)
```
I would like it changed to something more specific (but still passes both Linux and Windows versions).  My concern is that we not pass the new builtin gibberish.  Completely wildcarding the arguments defeats the purpose.

What do other reviewers think? @elizabethandrews @smanna12 ?

Comment (smanna12) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:13]: We can add one more RUN line with Windows triple so that we can check IR for both Linux and Windows platform. Does it sound good @premanandrao and @elizabethandrews?

Comment (srividya-sundaram) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp]: @premanandrao 

> My concern is that we not pass the new builtin gibberish.

Can you please elaborate what you mean by "`not pass the new builtin gibberish`"

LLVM IR for `printf` in **Linux** looks like : `@printf(ptr noundef nonnull dereferenceable(1) @.str, i32 noundef 24) #2`
  and in        **Windows** : `@printf(ptr noundef nonnull dereferenceable(1) @"??_C@_0L@JKDODDCB@hello?0?5?$CFd?6?$AA@", i32 noundef 24) #2`

Assuming you meant - `@"??_C@_0L@JKDODDCB@hello?0?5?$CFd?6?$AA@"` as gibberish, we still see that passed in  the IR. So not entirely sure what your comment means?

Also my understanding is that the string "hello"  is represented by the variable `.str` in Linux and Windows represents the same in a more complex(gibberish) variable name. ( you can see 'hello' embedded  in there)

I still implemented the suggestion because we could just wildcard the variable for 'hello' string and the rest of the stuff are the same for both Linux and Windows.

Comment (elizabethandrews) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp]: I believe @premanandrao means that by testing `{{.*}}` you're not actually testing what is passed to the builtin. Technically the current test not testing the content of `str` is problematic as well for the same reason and we should have caught that in our earlier review. I'm not sure how to fix it here though. Usually for IR checks (non-SYCL), we add a triple for CodeGen tests to ensure IR is generated for expected target. I'm not really sure how to do that for SYCL device runs though. Maybe we can restrict this test to run only on Linux ? @premanandrao please weigh in

Comment (elizabethandrews) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp]: Can you paste the Windows as a comment here?

Comment (elizabethandrews) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp]: Oh I just re-read @premanandrao's comment and  I see he doesn't think we need to test content of `str`.  If other reviewers are ok with this, I won't hold up the review.

Comment (premanandrao) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp]: > I believe @premanandrao means that by testing {{.*}} you're not actually testing what is passed to the builtin.

Yes.

> I see he doesn't think we need to test content of str. 

Yes.  :-)

File: clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp
Comment (premanandrao) [inline clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:13]: I am okay with this change if it passes on Windows as well.   I think it does, but I can't tell.

**Comment by pvchupin** [comment]
@premanandrao, @srividya-sundaram, is it ready to merge? approval is missing.

**Review by smanna12 — APPROVED** [review]

File: clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp
Comment (smanna12) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:13]: This will pass on windows i think:

LLVM IR for printf in Linux looks like : @printf(ptr noundef nonnull dereferenceable(1) @.str, i32 noundef 24) #2
and in Windows : @printf(ptr noundef nonnull dereferenceable(1) @"??_C@_0L@JKDODDCB@hello?0?5?$CFd?6?$AA@", i32 noundef 24) #2

Comment (premanandrao) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:13]: > This will pass on windows i think:

I think so too, but I have been bitten by accidental whitespace errors before.

Comment (smanna12) [reply clang/test/CodeGenSYCL/remove-restriction-builtin-printf.cpp:13]: I have approved it. Please wait for approval from @premanandrao.

**Review by elizabethandrews — APPROVED** [review]

**Pull request merged**
